msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt

```diff
@@ -221,8 +274,14 @@ class NativeAuthPublicClientApplicationKotlinTest : PublicClientApplicationAbstr
             MockApiResponseType.INVALID_PASSWORD
         )
 
+        val loggingNumber = if (allowPII) 114 else 114
```

Comment:
The number is the same regardless of the condition here

I'd also follow up on Sammy's feedback - the check is fragile, and not needed IMO. We shouldn't be testing the underlying behaviour of the logs, we should only check if the PII logging rules are valid.

I'm assuming what we want to check is: when `allowPII` is true, that messages containing PII on the allowed list are logged, and when `allowPII` is false, those messages are not logged.

In which case, you could add a boolean to the logger called something like `permittedPIIFound`. Since you're checking the `allowPIIFalseToCheck` list for permitted PII messages, you could set `permittedPIIFound` when a permitted message is logged. Then you could do `assertEqual(allowPII, permittedPIIFound)`. There's the possibility of this approach incorrectly failing if a flow doesn't log any messages in the permitted PII list, but we can assess if it works for the provided tests and update accordingly.

Reply:
Remove the logger callback + countdown mechanism because they are fragile and not scalable.

The assumption is right.  What we are using right now is verify().log() rather than assert(variable) or fail(). Based on your suggestions, I added allowList & disableList 
```
verify(externalLogger, never()).log(
            any(),
            any(),
            argThat(RegexMatcher(regex)),  // allowList items are logged but the containsPII should be true.
            eq(false)
        )
```